frontend/app/routes/_app+/letters+/_index.tsx

```diff
@@ -74,12 +74,12 @@ export default function LettersIndex() {
       </div>
       <ul className="divide-y border-y">
         {letters.map((letter) => {
-          const letterType = letterTypes.find(({ code }) => code === letter.name);
-          const letterName = letterType ? letterType[i18n.language === 'fr' ? 'nameFr' : 'nameEn'] : letter.name;
+          const letterTypeCd = letterTypes.find((letterType) => letterType.code == letter.name);
+          const letterTypeCode = { nameEn: letterTypeCd?.nameEn, nameFr: letterTypeCd?.nameFr };
           return (
             <li key={letter.id} className="py-4 sm:py-6">
               <Link reloadDocument to={`/letters/${letter.referenceId}/download`} className="font-medium hover:underline">
-                {letterName}
+                {getNameByLanguage(i18n.language, letterTypeCode)}
```

Comment:
Based on my previous suggestion.
```suggestion
                {letterName}
```

Reply:
Fixed: 04e0b73a6520ee0dc46d6b68ece420e1df40cca8